src/core/types.h

```diff
@@ -291,6 +293,7 @@ class SourceInfo {
     CFG* cfg;
     bool is_generator;
     std::string fn; // equivalent of code.co_filename
+    std::shared_ptr<LivenessAnalysis> liveness_info;
```

Comment:
oh now I remember why I didn't use a ```unique_ptr``` it doesn't work with forward declared types...